library/std/src/sys/pal/unix/fd.rs

```diff
@@ -269,6 +250,33 @@ impl FileDesc {
         }
     }
 
+    // We support old MacOS, iOS, watchOS, tvOS and visionOS versions that do not have `preadv`.
+    // There is no `syscall` possible in these platform for following apple versions:
+    // ios 14.0
+    // tvos 14.0
+    // macos 11.0
+    // watchos 7.0
+    // visionos 1.0
```

Comment:
Thinking about this... do we support visionOS before 1.0? I would think not? We could strong-link on visionOS if we don't. Arguably it makes the implementation messier, I suppose.

Reply:
I had to double check myself. [Looks like visionOS 1.0 is the first version](https://en.wikipedia.org/wiki/VisionOS#Release). In https://github.com/rust-lang/rust/pull/121419, the default [was set to 1.0](https://github.com/rust-lang/rust/blob/54692c3d0b873a5dc055f2f35081d0d6a2410a49/compiler/rustc_target/src/spec/base/apple/mod.rs#L384). This should be something easier to check though.

Reply:
After thinking about this more, I think that this comment is not to say "this is the minimum supported version", I think the original comment I paraphrased and amended meant "The following Operating system versions have this syscall: iOS 14.0, tvOS 14.0, macOS 11.0, watchOS 7.1".

Reply:
I've updated code comments to reflect my understanding of things and better describe it.